crates/cargo-wdk/src/actions/new/mod.rs

```diff
@@ -291,3 +291,476 @@ impl<'a> NewAction<'a> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        io::Error,
+        os::windows::process::ExitStatusExt,
+        path::{Path, PathBuf},
+        process::{ExitStatus, Output},
+    };
+
+    use clap_verbosity_flag::Verbosity;
+
+    use super::error::NewActionError;
+    use crate::{
+        actions::{new::NewAction, DriverType},
+        providers::{
+            error::{CommandError, FileError},
+            exec::MockCommandExec,
+            fs::MockFs,
+        },
+    };
+
+    // Helper struct for setting up different failure scenarios in the
+    // `update_cargo_toml` step.
+    struct UpdateCargoTomlCaseSetupHelper {
```

Comment:
```suggestion
    struct UpdateCargoTomlTestCase {
```

What this struct seems to hold is pre-conditions and post-conditions of a test. So a good name for it is just `*TestCase`. 

You don't really need a struct for this. Just use an unnamed tuple like you have done on line 330.

Reply:
This lint seems to be about type nesting. Can you try creating a type alias for the function pointer type and use that alias in the tuple? That might stop the warning.